### ISSUE TITLE
Switch to using Dokka for API documentation

### ISSUE THREAD

**Issue by cammace** [mapbox/android-docs#244]
With Kotlin, a new documentation process has been introduced [called Dokka](https://github.com/Kotlin/dokka) which looks to be the future of Java and Kotlin documentation. We can use a Gradle script to output HTML, Markdown, or other supported outputs which represent the Javadoc. My understanding is that we won't need to change the javadoc written in the code itself but instead just replace the Gradle script and figure out the best way moving forward on how/where to host this new API documentation format. Thinking I might explore this in Mapbox Java 3.0 first, see how it goes and what the output looks like. In any case, this looks a hell of a lot better than Javadoc which is ancient.

cc: @mapbox/android @danswick

**Comment by danswick** [comment]
![oh my god](http://media4.giphy.com/gifsu/26FeXAzRypPyoQ1BC/giphy-caption.gif)

**Comment by langsmith** [comment]
fyi, I've begun work on a branch for trying Dokka with the Navigation SDK for Android Javadocs. More to come.

https://github.com/mapbox/mapbox-navigation-android/compare/ls-switch-to-dokka-for-javadocs

**Comment by zugaldia** [comment]
@langsmith Any juicy screenshots from that branch that you could share?

**Comment by langsmith** [comment]
Nah, unfortunately, I'm not close to figuring out how to set this up. Not too familiar with Gradle plugins, so the experimentation and growing pains continue.
